federation/pkg/federation-controller/configmap/configmap_controller_test.go

```diff
@@ -143,11 +143,18 @@ func TestConfigMapController(t *testing.T) {
 	}
 
 	configmapWatch.Modify(configmap1)
-	updatedConfigMap = GetConfigMapFromChan(cluster1UpdateChan)
-	assert.NotNil(t, updatedConfigMap)
-	assert.Equal(t, configmap1.Name, updatedConfigMap.Name)
-	assert.Equal(t, configmap1.Namespace, updatedConfigMap.Namespace)
-	assert.True(t, util.ConfigMapEquivalent(configmap1, updatedConfigMap))
+	for {
```

Comment:
It happens because we are running controller with greatly reduced delay time. It is possible that the change doesn't get through the async mock transport layer before the config map is  processed again. We could probably rethink the mocks but double delivery in this type of integration test with reduced delays will still be possible.